src/infer_context.cc

```diff
@@ -145,36 +167,45 @@ InferContext::SendRequest(
 
     total_ongoing_requests_++;
   } else {
-    std::chrono::time_point<std::chrono::system_clock> start_time_sync,
-        end_time_sync;
-    thread_stat_->idle_timer.Start();
-    start_time_sync = std::chrono::system_clock::now();
     cb::InferResult* results = nullptr;
-    thread_stat_->status_ = infer_backend_->Infer(
-        &results, *(infer_data_.options_), infer_data_.valid_inputs_,
-        infer_data_.outputs_);
+    thread_stat_->idle_timer.Start();
+    auto start_time_sync = std::chrono::system_clock::now();
+    if (streaming_) {
+      thread_stat_->status_ = infer_backend_->StreamInfer(
+          &results, *(infer_data_.options_), infer_data_.valid_inputs_,
+          infer_data_.outputs_);
+    } else {
+      thread_stat_->status_ = infer_backend_->Infer(
+          &results, *(infer_data_.options_), infer_data_.valid_inputs_,
+          infer_data_.outputs_);
+    }
     thread_stat_->idle_timer.Stop();
+
+    std::vector<std::chrono::time_point<std::chrono::system_clock>>
+        response_timestamps{std::chrono::system_clock::now()};
     RequestRecord::ResponseOutput response_outputs{};
+
     if (results != nullptr) {
       if (thread_stat_->status_.IsOk()) {
         response_outputs = GetOutputs(*results);
         thread_stat_->status_ = ValidateOutputs(results);
+
+        // When streaming, use the timestamps collected from client
+        if (streaming_) {
+          results->ResponseTimestamps(&response_timestamps);
+        }
```

Comment:
I still feel a bit strange about how each received message of a response (i.e. calls to CompletionQueue::Next() until `ok` is false) has its own timestamp, but all of the messages sent for an outgoing request (i.e. calls to Write()) don't have their own timestamps.

Did the stakeholders express what kind of benchmark metrics/statistics they want?

Reply:
Removed this part since stakeholders currently only care about the request latency and throughput (not the fine-grained collection of timestamps of messages received in a response).

Reply:
But how do they want latency and throughout defined? Do they want the time between the first outgoing message of a single complete inference and the last incoming message of the same single inference to be considered one latency value in the benchmark?

Reply:
They were fine with whatever the default way GAP/PA was measuring for request latency and throughput.

Reply:
I guess my question is more for us. Can we confirm how GAP/PA is defining latency for this bidirectional streaming feature?

Reply:
The main reason why I added timestamps for response was because async stream case was doing it and I thought it would be a good to have parity. But it was not a requirement 😅 I was going a bit of ahead of myself.

Reply:
Oh so did you remove the set of timestamps associated with each incoming message of a single inference?

It's now just one timestamp for when the last message of the single inference came in?

Reply:
Now it's just the same as how we used to do for plain synchronous inference call: before & after the InferContext::Infer call (https://github.com/triton-inference-server/perf_analyzer/blob/f53136ae2a2c9be5c0eba4b1ab0fe81f953182df/src/infer_context.cc#L150-L156).